app/models/user.rb

```diff
@@ -10,4 +12,32 @@ class User < ApplicationRecord
   validates :password, presence: true, length: { minimum: 6 }
 
   has_secure_password
+
+  # Returns the hash digest of the given string.
+  def User.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+               BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  # Returns a random token.
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
```

Comment:
* `self.` is not necessary inside an instance method.
  * Overall I don't think `digest`, `new_token` and any other methods making direct use of `BCrypt`/`SecureRandom ` belong in the `User` model which shouldn't have to concern itself with low-level authentication details. I'd suggest creating an authentication helper/service to encapsulate these details and rather have the `User` model call that. That gives you flexibility for making auth mechanism changes without touching user model & has added bonus of simplifying testing :)